core/dbt/contracts/graph/manifest.py

```diff
@@ -1228,13 +1229,27 @@ def compatible_previous_versions(self):
             ("manifest", 8),
         ]
 
+    @classmethod
+    def upgrade_schema_version(cls, data):
```

Comment:
Hmm I got turned around because the function's name led me to believe that the schema's version is upgraded, but I don't think of `from_dict` alone as an upgrading action. Is it generally expected that the verb "upgrade" for dbt-core functions means "change if changes are needed"? Additionally does `from_dict` actually do an upgrade of some sort?

Reply:
The part that does the "upgrade" is "upgrade_manifest_json". That code implements any changes that are necessary to have an older manifest schema version loaded by the current manifest code.  "from_dict" is used all over the codebase and is implemented by mashumaro, which is pulled in by dbtClassMixin. It just instantiates an object from a dictionary.

Reply:
Gotcha! Thank you for walking me through that 🙂